src/lib/markbind/src/utils/urls.js

```diff
@@ -8,48 +8,59 @@ const {
   BOILERPLATE_FOLDER_NAME,
 } = require('../constants');
 
+
 /**
+ * Returns the absolute path of of the immediate parent site of the specified file path.
  * @param filePath The absolute file path to look up that is nested inside the root directory
  * @param root The base directory from which to terminate the look up
  * @param lookUp The set of urls representing the sites' base directories
- * @return An object containing
- *         1. The parent site's path of that the immediate parent site of the specified filePath,
- *            or the root site's path if there is none
- *         2. The relative path from (1) to the immediate parent site of the specified filePath
+ * @param needRelative Whether the relative file path from the root site to the immediate parent site
+ *                     should be returned as well. Defaults to false if unspecified
+ * @return Either
```

Comment:
Is it possible/inexpensive enough for us to just calculate and return both the absolute and relative path every time? I think having to deal with two possible return types make this function more complicated, and can easily lead to errors especially without a type system in place.

Reply:
I went with two possible return types to avoid e.g. `{ absolute: parentSite } = urlUtils.calculate...` all over. I merged it into one now though, will revert if preferred!

Reply:
I personally would lean towards avoiding hard to spot errors over having slightly shorter code, but I'm open to the input of others.

What do you think about splitting them up into two functions? Would help make this more explicit while reducing unnecessary destructuring at the call site? (i.e. `calculateAbsoluteParentSitePath` and `calculateAbsoluteAndRelativeParentSitePaths`)

Reply:
Sorry for the late reply! I've changed it back to two functions